### PR TITLE
Gateway API: refactoring and handle unmanaged-gateway class status, adopt 0.8.0

### PR DESCRIPTION
**Pull request by howardjohn**
In https://github.com/istio/istio/pull/46152, support for Routability was added. This was then removed from the API.
This includes a fix to write `unmanaged-gateway` GatewayClass status

### CODE REVIEW

**Review by keithmattix — COMMENTED** [review]
Just one question around the service type annotation

File: manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml [hunk 1]
```diff
@@ -317,8 +317,8 @@ spec:
   {{- end }}
   selector:
     istio.io/gateway-name: {{.Name}}
-  {{- if .Spec.Addresses }}
+  {{- if and (.Spec.Addresses) (eq .ServiceType "LoadBalancer") }}
   loadBalancerIP: {{ (index .Spec.Addresses 0).Value | quote}}
   {{- end }}
-  type: {{ index .Annotations "networking.istio.io/service-type" | default "LoadBalancer" | quote }}
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml:323]: https://github.com/istio/istio/pull/46368/files#diff-c49c9e7fbae41236bcd0daff3ba0f8d2be118f9d4d33086ac91880ff653f7a4dR319-R322 handles it still, or is supposed to. I think we have a test, let me double check.

Although we warn  to use "Spec.Infrastructure.Routeability" still which is a bug (I missed due to the typo :-( )

[the remark this answers is not in the document]
Comment (keithmattix) [reply manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml:323]: Ok I think that's fair. Make the template dumb and handle more on the controller end. I can buy that 👍🏾

**Review by keithmattix — APPROVED** [review]
Approving with a hold in case you want to fix the warning to use routability

**Review by ericvn — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]

**Pull request merged**